src/arrayMethodJoin.js

```diff
@@ -4,8 +4,29 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
+  [].__proto__.join2 = function(separator = ',') {
     // write code here
+    const arr = [...this];
```

Comment:
Creating a new array 'arr' is unnecessary. You can directly use 'this' to access the original array and avoid the extra copy. 🤔